packages/connect/src/create-store.js

```diff
@@ -19,36 +21,49 @@ const getSnapshot = obj => {
   }
 };
 
-const convertToAction = (fn, instance) => (...args) => {
-  const first = fn(instance);
-  if (first instanceof Promise)
-    return new Promise(resolve => first.then(() => resolve()));
-  if (typeof first === "function") {
-    const second = first(...args);
-    if (second instanceof Promise)
-      return new Promise(resolve => second.then(() => resolve()));
+const convertToAction = (fn, instance, context) => (...args) => {
+  let action = observable(fn, null, context);
+  action = fn(instance);
+
+  let runMiddlewares = compose(middlewares);
+  runMiddlewares(context, instance, instance.actions)
+    .then(() => {})
+    .catch(err => console.error(err));
+
+  if (action instanceof Promise)
+    return new Promise(resolve => action.then(() => resolve()));
+  if (typeof action === "function") {
+    const actionWithParams = action(...args);
+    if (actionWithParams instanceof Promise)
+      return new Promise(resolve => actionWithParams.then(() => resolve()));
   }
 };
 
-const convertedActions = (obj, instance) => {
-  if (typeof obj === "function") return convertToAction(obj, instance);
+const convertedActions = (obj, instance, context) => {
+  if (typeof obj === "function") return convertToAction(obj, instance, context);
   else if (obj instanceof Object) {
     return Object.keys(obj).reduce((newObj, key) => {
-      newObj[key] = convertedActions(obj[key], instance);
+      newObj[key] = convertedActions(obj[key], instance, context);
       return newObj;
     }, {});
   }
 };
 
-export const createStore = store => {
-  const observableState = observable(store.state);
+export const createStore = (store, context) => {
+  // TODO: onl throw in dev ?
```

Comment:
We can later add a `__DEV__` mode to `connect` to make use of those warnings. React does something similar, they have many warnings like: 
```js
if (__DEV__) { throw new Error(`Terrible error`) } 
```
and use a webpack (or rollup, not sure) plugin to remove those if statements completely for production.